src/main/java/datawave/query/util/IndexedDatesValue.java

```diff
@@ -0,0 +1,196 @@
+package datawave.query.util;
+
+import org.apache.accumulo.core.data.Value;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.BitSet;
+import java.util.Iterator;
+import java.util.TreeSet;
+
+public class IndexedDatesValue {
+    
+    private static final Logger log = LoggerFactory.getLogger(IndexedDatesValue.class);
+    
+    private YearMonthDay startDay;
+    private TreeSet<YearMonthDay> indexedDatesSet = new TreeSet<>();
```

Comment:
I thought we were going to internalize this as a set of ranges instead of distinct YearMonthDays.  Doing it this way will create a ton of objects.  Also, we need to figure out how to deal with that fact that data is indexed already but has no bits set.  So either we need a way of updating legacy metadata to presume that it has been indexed since the beginning of time, or we assume that we have indexed data from the beginning of time until the first bit set.

Reply:
I am thinking the storage of a bitset is excellent, we simply should transform that into a set of FieldIndexHoles instead of a large set of YearMonthDay objects directly.